astrolabe/runner.py

```diff
@@ -91,29 +93,37 @@ def is_cluster_state(self, goal_state):
         cluster_info = self.cluster_url.get().data
         return cluster_info.stateName.lower() == goal_state.lower()
 
-    def verify_cluster_configuration_matches(self, state):
+    def verify_cluster_configuration_matches(self, expected_configuration):
         """Verify that the cluster config is what we expect it to be (based on
         maintenance status). Raises AssertionError."""
-        state = state.lower()
-        if state not in ("initial", "final"):
-            raise AstrolabeTestCaseError(
-                "State must be either 'initial' or 'final'.")
         cluster_config = self.cluster_url.get().data
         assert_subset(
             cluster_config,
-            self.spec.maintenancePlan[state].clusterConfiguration)
+            expected_configuration.clusterConfiguration)
         process_args = self.cluster_url.processArgs.get().data
         assert_subset(
-            process_args, self.spec.maintenancePlan[state].processArgs)
+            process_args, expected_configuration.processArgs)
 
-    def initialize(self):
+    def initialize(self, no_create=False):
         """
         Initialize a cluster with the configuration required by the test
         specification.
         """
+        
+        if no_create:
+            try:
+                self.cluster_url.get().data
+                # If --no-create was specified and the cluster exists, skip
+                # initialization. If the cluster does not exist, continue
+                # with normal creation.
+                return
```

Comment:
Is there any concern that a previous run might have left the cluster in an unhealthy state if it was terminated unexpectedly? If the cluster is found to be running, would it make sense to try and assert that its current state matches the expected initialConfiguration?

Reply:
I added a check for cluster configuration and if it differs from the one specified in the scenario, the cluster will be reconfigured.